packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -352,48 +352,74 @@ typedef bool RoutePredicate(Route<dynamic> route);
 /// The app's home page route is named '/' by default and other routes are
 /// given pathnames by convention.
 ///
-/// ### Defining a popup route
+/// ### Routes can return a value
+///
+/// When a route is pushed to ask the user for a value, the value can be
+/// returned via the [pop] method's result parameter.
+///
+/// Methods that push a route return a Future. The Future resolves when
+/// the route is popped and the Future's value is the [pop] method's result
+/// parameter.
+///
+/// For example if we wanted to ask the to press 'OK' to confirm an
+/// operation we could await the result of [Navigator.push]:
+///
+/// ```
+/// bool value = await Navigator.of(context).push(new MaterialPageRoute<bool>(
+///   builder: (BuildContext context) {
+///     return new Center(
+///       child: new FlatButton(
```

Comment:
I did try it within a MaterialApp/Scaffold. I thought it was OK to assume as much here, since the previous example creates a MaterialApp and I'm creating a MaterialPageRoute.

I wish I could link to complete flutter projects that contain the snippets being shared here. I wish I could click on an example link and find myself in an IDE window that contained the project, ready to run. I wish.